sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1991,6 +1991,22 @@ object functions {
   def tan(columnName: String): Column = tan(Column(columnName))
 
   /**
+   * Computes the cotangent of the given column.
+   *
+   * @group math_funcs
+   * @since 2.2.0
+   */
+  def cot(e: Column): Column = withExpr { Cot(e.expr) }
+
+  /**
+   * Computes the cotangent of the given column name.
+   *
+   * @group math_funcs
+   * @since 2.2.0
+   */
+  def cot(colName: String): Column = cot(Column(colName))
+
+  /**
```

Comment:
I'm not sure though, do we strictly need to add this function in this file? If we add this in `FunctionRegistry`, you know we can use this thru `selectExpr`. ISTM this file seems to include frequently-used functions only. cc: @gatorsmile